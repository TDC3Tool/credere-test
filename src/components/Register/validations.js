export const requiredFiled = val => (!val || !val.length ? 'Campo obrigat√≥rio' : '')
