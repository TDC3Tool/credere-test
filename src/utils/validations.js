export const requiredField = val => (!val || !val.length ? 'Campo obrigat√≥rio' : '')
