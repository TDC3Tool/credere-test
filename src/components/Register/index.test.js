import { requiredFiled } from './validations'

describe('Register Component', () => {
  it('should be validation requiredField is defined', () => {
    const expected = requiredFiled()
    expect(expected).toBeDefined()
  })

  it('should return a empty error description', () => {
    const expected = requiredFiled('my value')
    expect(expected).toBe('')
  })

  it('should return a error description', () => {
    const expected = requiredFiled('')
    expect(expected).toBe('Campo obrigat√≥rio')
  })
})
