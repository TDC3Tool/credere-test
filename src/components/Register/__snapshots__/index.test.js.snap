// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register Component should render partially components of Register 1`] = `ReactWrapper {}`;

exports[`Register Component should render register component 1`] = `
<form
  className="sc-dnqmqq dkkdNN"
  noValidate={true}
  onSubmit={[Function]}
>
  <div>
    <h3
      className="user-title sc-bwzfXH jmxchW"
    >
      Cliente
    </h3>
    <div
      className="grid user-details "
    >
      <div
        className="border-sizing sc-EHOje hcPLoZ"
      >
        <label
          className="sc-bZQynM fGYiaS"
          htmlFor="name"
        >
          <span
            className="sc-bxivhb bUZGRr"
          >
            Nome
          </span>
           *
        </label>
        <input
          className="sc-gzVnrw fMAtCm"
          id="name"
          name="name"
          onChange={[Function]}
          required={true}
          type="text"
        />
        
      </div>
      <div
        className="border-sizing sc-EHOje hcPLoZ"
      >
        <label
          className="sc-bZQynM fGYiaS"
          htmlFor="birthday"
        >
          <span
            className="sc-bxivhb bUZGRr"
          >
            Data de nascimento
          </span>
           *
        </label>
        <input
          className="sc-gzVnrw fMAtCm"
          id="birthday"
          max="2019-07-20"
          name="birthday"
          onChange={[Function]}
          required={true}
          type="date"
        />
        
      </div>
    </div>
    <div
      className="grid user-address"
    >
      <div
        className="border-sizing sc-EHOje hcPLoZ"
      >
        <label
          className="sc-bZQynM fGYiaS"
          htmlFor="state"
        >
          <span
            className="sc-bxivhb bUZGRr"
          >
            Estado
          </span>
           *
        </label>
        <input
          className="sc-gzVnrw fMAtCm"
          id="state"
          maxLength="2"
          name="state"
          onChange={[Function]}
          pattern="[A-Za-z]{2}"
          required={true}
          type="text"
        />
        
      </div>
    </div>
  </div>
  <div
    className="sc-gZMcBi geStiv"
  >
    <h3
      className="phones-title sc-bwzfXH jmxchW"
    >
      Telefones
    </h3>
    <div
      className="grid phones-list"
    >
      <span
        className="sc-bxivhb bUZGRr"
      >
        Não há telefones cadastrados!
      </span>
    </div>
    <div
      className="grid phones-form"
    >
      <div
        className="flex"
      >
        <button
          className="button-md button-outline-success transition-all sc-ifAKCX flbtib"
          onClick={[Function]}
          type="button"
        >
          Adicionar mais
        </button>
      </div>
    </div>
  </div>
  <div
    className="sc-gqjmRU loubtk"
  >
    <h3
      className="emails-title sc-bwzfXH jmxchW"
    >
      E-mails
    </h3>
    <div
      className="grid emails-list"
    >
      <span
        className="sc-bxivhb bUZGRr"
      >
        Não há e-mails cadastrados!
      </span>
    </div>
    <div
      className="grid emails-form"
    >
      <div
        className="flex"
      >
        <button
          className="button-md button-outline-success transition-all sc-ifAKCX flbtib"
          onClick={[Function]}
          type="button"
        >
          Adicionar mais
        </button>
      </div>
    </div>
  </div>
  <div
    className="sc-iwsKbI leChZW"
  >
    <h3
      className="parent-title sc-bwzfXH jmxchW"
    >
      Responsável
    </h3>
    <div
      className="grid parent-details"
    >
      <div
        className="border-sizing sc-EHOje hcPLoZ"
      >
        <label
          className="sc-bZQynM fGYiaS"
          htmlFor="parent-name"
        >
          <span
            className="sc-bxivhb bUZGRr"
          >
            Nome
          </span>
           *
        </label>
        <input
          className="sc-gzVnrw fMAtCm"
          id="parent-name"
          name="parent-name"
          onChange={[Function]}
          required={true}
          type="text"
        />
        
      </div>
      <div
        className="grid phones-form-fields"
      >
        <div
          className="border-sizing sc-EHOje hcPLoZ"
        >
          <label
            className="sc-bZQynM fGYiaS"
            htmlFor="parent-phone-code"
          >
            <span
              className="sc-bxivhb bUZGRr"
            >
              DDD
            </span>
             *
          </label>
          <input
            className="sc-gzVnrw fMAtCm"
            id="parent-phone-code"
            maxLength="2"
            minLength="2"
            name="parent-phone-code"
            onChange={[Function]}
            pattern="[0-9]*"
            placeholder="DDD"
            required={true}
            type="text"
          />
          
        </div>
        <div
          className="border-sizing sc-EHOje hcPLoZ"
        >
          <label
            className="sc-bZQynM fGYiaS"
            htmlFor="parent-phone-number"
          >
            <span
              className="sc-bxivhb bUZGRr"
            >
              Telefone
            </span>
             *
          </label>
          <input
            className="sc-gzVnrw fMAtCm"
            id="parent-phone-number"
            maxLength="9"
            minLength="8"
            name="parent-phone-number"
            onChange={[Function]}
            pattern="[0-9]*"
            required={true}
            type="text"
          />
          
        </div>
      </div>
    </div>
  </div>
  <button
    className="button-lg button-normal-default transition-all sc-ifAKCX flbtib"
    onClick={[Function]}
    type="submit"
  >
    Salvar Cliente
  </button>
</form>
`;
